Wave1_data/4.processing_profiled_features/scripts/3.normalize_sc_across_time.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Normalize annotated single cells using negative control
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+import pprint
+
+import pandas as pd
+from pycytominer import normalize
+from pycytominer.cyto_utils import output
+
+
+# ## Set paths and variables
+
+# In[2]:
+
+
+# directory where combined parquet file are located
+data_dir = pathlib.Path("../data/annotated_data_combined")
+
+# directory where the normalized parquet file is saved to
+output_dir = pathlib.Path("../data/normalized_data")
+output_dir.mkdir(exist_ok=True)
+
+
+# ## Define dict of paths
+
+# In[3]:
+
+
+# dictionary with each run for the cell type
+dict_of_inputs = {
+    "live_cell_pyroptosis_wave1_sc": {
+        "combined_file_path": pathlib.Path(
+            f"{data_dir}/live_cell_pyroptosis_wave1_sc.parquet"
+        ).resolve(strict=True),
+        "output_file_path": pathlib.Path(
+            f"{output_dir}/live_cell_pyroptosis_wave1_sc_pan_time_norm.parquet"
+        ).resolve(),
+    },
+}
+
+
+# ## Normalize with standardize method with negative control on annotated data
+
+# The normalization needs to occur per time step. 
+# This code cell will split the data into time steps and normalize each time step separately.
+# Then each normalized time step will be concatenated back together. 
+
+# In[4]:
+
+
+for info, input_path in dict_of_inputs.items():
+    # read in the annotated file
+    print(input_path)
+    annotated_df = pd.read_parquet(input_path["combined_file_path"])
+    annotated_df.reset_index(drop=True, inplace=True)
+    # Normalize the single cell data per time point
```

Comment:
This code comment seems to maybe be misplaced. I don't see any filtering or separation based on time. Consider removing or updating.

Reply:
I need to make a figure describing these approaches